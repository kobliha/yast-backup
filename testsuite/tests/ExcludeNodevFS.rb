# encoding: utf-8

# Backup module - testsuite
#
# ExcludeNodevFS function test
#
# testedfiles: Backup.ycp
#
# $Id$
#
module Yast
  class ExcludeNodevFSClient < Client
    def main
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.import "Pkg" # override packamanager

      Yast.import "Backup"

      @readmap = {}
      Testsuite.Test(lambda { Backup.ExcludeNodevFS }, [@readmap, {}, @execmap], nil)
      Testsuite.Dump(Backup.fs_exclude)

      @readmap = {
        "proc" => {
          "filesystems" => {
            "afs"      => "nodev",
            "iso9660"  => "\t",
            "minix"    => "\t",
            "nfs"      => "nodev",
            "pipefs"   => "nodev",
            "proc"     => "nodev",
            "ramfs"    => "nodev",
            "reiserfs" => "\t",
            "rootfs"   => "nodev"
          }
        }
      }
      Testsuite.Test(lambda { Backup.ExcludeNodevFS }, [@readmap, {}, @execmap], nil)
      Testsuite.Dump(Backup.fs_exclude)

      @readmap = {}
      Backup.fs_exclude = []
      Testsuite.Test(lambda { Backup.ExcludeNodevFS }, [@readmap, {}, @execmap], nil)
      Testsuite.Dump(Backup.fs_exclude)

      nil
    end
  end
end

Yast::ExcludeNodevFSClient.new.main
