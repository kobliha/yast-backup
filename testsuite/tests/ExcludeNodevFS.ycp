// Backup module - testsuite
//
// ExcludeNodevFS function test
//
// $Id$
//

{

include "testsuite.ycp";

map readmap = $[ "target" : $["string" : "", "size" : -1 ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "cdrom" : $["manual" : []]]  ];

TESTSUITE_INIT([ readmap, [], [] ], nil);

import "Pkg";   // override packamanager

import "Backup";

readmap = $[];
TEST( ``(Backup::ExcludeNodevFS()), [ readmap, $[], $[] ], nil );
DUMP(Backup::fs_exclude);

readmap = $[ "proc" : $[ "filesystems" : $["afs":"nodev", "iso9660":"\t", "minix":"\t", "nfs":"nodev", "pipefs":"nodev", "proc":"nodev", "ramfs":"nodev", "reiserfs":"\t", "rootfs":"nodev"]] ];
TEST( ``(Backup::ExcludeNodevFS()), [ readmap, $[], $[] ], nil );
DUMP(Backup::fs_exclude);

readmap = $[];
Backup::fs_exclude = [];
TEST( ``(Backup::ExcludeNodevFS()), [ readmap, $[], $[] ], nil );
DUMP(Backup::fs_exclude);

}

