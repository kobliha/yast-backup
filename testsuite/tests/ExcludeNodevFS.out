Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Read	.target.tmpdir "/tmp"
Read	.target.size "/tmp/autoinst.xml" -1
Dir	.autoinst.menuentry.s: []
Read	.proc.filesystems nil
Return	nil
Dump	["iso9660", "ntfs"]
Read	.proc.filesystems $["afs":"nodev", "iso9660":"\t", "minix":"\t", "nfs":"nodev", "pipefs":"nodev", "proc":"nodev", "ramfs":"nodev", "reiserfs":"\t", "rootfs":"nodev"]
Return	nil
Dump	["afs", "iso9660", "nfs", "ntfs", "pipefs", "proc", "ramfs", "rootfs"]
Read	.proc.filesystems nil
Return	nil
Dump	[]
